### ISSUE TITLE
readline segfaults with python 3.7.2

### ISSUE THREAD

**Issue by jlquinn** [ludwigschwardt/python-readline#1]
Hi, I discovered the rename the hard way.  I install python 3.7.2 on Ubuntu 16.04 and installed readline through pip3.  Python now segfaults.  I traced it down to the readline pip package.  I was able to find that gnureadline is the replacement.

Please prevent readline from matching recent python because of the segfault problem.  It definitely breaks 3.7.2 and I see reports that 3.6 breaks as well.  This will help prevent surprises.  Thanks!

**Comment by ludwigschwardt** [comment]
Good idea. Just remember that there is absolutely no reason to install readline or gnureadline on Ubuntu, since Linux Pythons typically ship with readline included (unless you are using ActiveState Python or some other non-standard distribution - I'm curious to know your setup :-)).

**Comment by jlquinn** [comment]
I'm wanting to use python 3.7 on Ubuntu 16.04.  The default version is 3.5, which is why I'm building my own python.

**Comment by ludwigschwardt** [comment]
Ah, good point. Why do you need readline separately though? Readline 6.3 should be available on 16.04, which is the same version embedded in the gnureadline package.

**Comment by jlquinn** [comment]
Ah, perhaps I'm confused.  On python 2.7 I had to install the readline pip package because without that, there was no readline support.  I assumed the same was true for python 3.7.  Am I missing something?

**Comment by ludwigschwardt** [comment]
If you are building Python from the source tarballs provided at python.org, it might be that you are missing the headers / include files for optional dependencies like readline, zlib, etc.

For example, on Ubuntu 16.04 you would want to do (see the Python [developer's guide](https://devguide.python.org/setup/#build-dependencies) for more info)
```
sudo apt-get build-dep python3.5
```
since the dependencies have not changed (or very little) from 3.5 to 3.7.

Specifically, for readline support you need the `libreadline-dev` Ubuntu package to be installed *before* building Python from source.

Readline support in this case can be added afterwards via the readline package, but that is not the purpose of this package. It is meant for systems that lack readline completely or has an incompatible replacement, while still being Posix compliant. That pretty much narrows it down to macOS... Although ActiveState Python on Linux also benefits from this due to its licensing restrictions.

**Comment by jlquinn** [comment]
Thank you.  This makes a lot more sense now.

**Comment by ludwigschwardt** [comment]
Glad to be of help! I'll still take up this issue though - no need to crash knowingly.
